sql/src/main/java/org/apache/druid/sql/calcite/external/Externals.java

```diff
@@ -242,7 +242,11 @@ private static String convertType(String name, SqlDataTypeSpec dataType)
     if (typeName == null || !typeName.isSimple()) {
       throw unsupportedType(name, dataType);
     }
-    SqlTypeName type = SqlTypeName.get(typeName.getSimple());
+    String simpleName = typeName.getSimple();
+    if (StringUtils.toLowerCase(simpleName).startsWith(("complex<"))) {
+      return simpleName;
+    }
+    SqlTypeName type = SqlTypeName.get(simpleName);
```

Comment:
nit: i know this isn't new, but these variable names are kinda confusing which initially had me wondering why we were looking up the type name of the type name 😅 but see now that `typeName` is an `SqlIdentifier` and `type` is an `SqlTypeName`.

Reply:
Changed all the names. I hope the new names are a bit more clear.